oak-query-spi/src/main/java/org/apache/jackrabbit/oak/spi/query/QueryIndex.java

```diff
@@ -491,6 +510,17 @@ public IndexPlan build() {
                     private final boolean deprecated =
                             Builder.this.deprecated;
                     private final boolean logWarningForPathFilterMismatch = Builder.this.logWarningForPathFilterMismatch;
+                    private final Map<Level, List<String>> additionalMessages = Builder.this.additionalMessages;
+
+                    private String getAdditionalMessageString() {
+                        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
I don't think this is completely safe. If there are no additional messages, the StringBuilder would call `deleteCharAt` with a negative value. You could make it more concise and functional using streams

```java
private String getAdditionalMessageString() {
    return additionalMessages.entrySet().stream()
            .map(e -> e.getKey() + " : " + e.getValue())
            .collect(Collectors.joining(", "));
}
```

Reply:
Fixed. thanks!